allennlp/data/token_indexers/pretrained_transformer_indexer.py

```diff
@@ -70,22 +70,30 @@ def _add_encoding_to_vocabulary_if_needed(self, vocab: Vocabulary) -> None:
         if self._added_to_vocabulary:
             return
 
-        vocab_field_name = None
-        if hasattr(self._tokenizer, "vocab"):
-            vocab_field_name = "vocab"
-        elif hasattr(self._tokenizer, "encoder"):
-            vocab_field_name = "encoder"
+        # Find the key under which the vocab is hiding.
+        for vocab_field_name in ["vocab", "encoder", "sp_model"]:
+            if hasattr(self._tokenizer, vocab_field_name):
+                break
+        else:
```

Comment:
This for-else might be a little weird. Maybe something like
```
a = None
for b in list:
  if cond:
    a = x
```

Reply:
Then I'd need an extra variable.